deployment/k8s/vault/vault_secret_store.yml

```diff
@@ -0,0 +1,15 @@
+apiVersion: external-secrets.io/v1beta1
+kind: SecretStore
+metadata:
+  name: vault-backend
+spec:
+  provider:
+    vault:
+      server: "http://vault.vault.svc.cluster.local:8200"
+      path: "secret"
+      version: "v2"
+      auth:
+        tokenSecretRef:
+          name: "vault-token"
+          key: "token"
+
```

Comment:
external secrets is bunch of CRs that enable integration with various backend, here we are setting our provider to vault and provide basic auth that will allow this store to sync secrets from vault